nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -87,7 +87,9 @@ public void refresh() throws NessieNotFoundException {
   }
 
   public NessieIcebergClient withReference(String requestedRef, String hash) {
-    if (null == requestedRef) {
+    if (null == requestedRef ||
+        getRef().getReference().getName().equals(requestedRef) ||
+        getRef().getHash().equals(hash)) {
```

Comment:
Sure that you want equals on ref-name _OR_ hash? I.e. still return `this`, if the branch name doesn't change?

Reply:
at this point yes, because Trino will make heavier use of this method and we don't want to do another REST call to fetch the branch if it didn't change

Reply:
> at this point yes, because Trino will make heavier use of this method and we don't want to do another REST call to fetch the branch if it didn't change

but functionally it is wrong in case of "drop a reference and create new reference with same name but different hash/source"

Can we make trino pass the hash as well ?